libeos-payg/efi.c

```diff
@@ -537,26 +544,60 @@ efivarfs_read (const char  *name,
 
 /* eospayg_efi_secureboot_active:
  *
- * Check if the system was booted via SecureBoot
+ * Check if the system was booted via real SecureBoot in result
  *
  * Returns: %TRUE if booted via SecureBoot, %FALSE otherwise
  */
 gboolean
 eospayg_efi_secureboot_active (void)
 {
-  g_autofree char *tname = full_efi_name (GLOBAL_VARIABLE_GUID, "SecureBoot");
-  g_autofree unsigned char *content = NULL;
-  int size;
+  g_return_val_if_fail (efi != NULL, FALSE);
 
-  /* In test mode let's pretend secure boot is enabled */
-  if (test_mode)
-    return TRUE;
+  g_autofree char *name = full_efi_name (GLOBAL_VARIABLE_GUID, "SecureBoot");
 
-  content = efivarfs_read (tname, &size, NULL);
-  if (!content || size != 1)
-    return FALSE;
+  return eospayg_efi_var_read_fullname_boolean (name, NULL);
+}
 
-  return !!content[0];
+/* eospayg_efi_setupmode_active:
+ *
+ * Check if the system was booted with SetupMode
+ *
+ * Returns: %TRUE if booted via SetupMode, %FALSE otherwise
+ */
+gboolean
+eospayg_efi_setupmode_active (void)
+{
+  g_return_val_if_fail (efi != NULL, FALSE);
+
+  g_autofree char *name = full_efi_name (GLOBAL_VARIABLE_GUID, "SetupMode");
+
+  return eospayg_efi_var_read_fullname_boolean (name, NULL);
+}
+
+/* eospayg_efi_secureboot_setup_active:
+ *
+ * Check if the system was booted with SecureBoot Setup config.
+ *
+ * However, some systems such as OLPC LEAP W502 do no have the SecureBootSetup
+ * EFI variable.
+ *
+ * Returns: one of the enum. %EFIVAR_NOT_EXIST if SecureBootSetup variable does
+ *          not exist, %EFIVAR_TRUE if system boots with set SecureBootSetup,
+ *          %EFIVAR_FALSE otherwise
+ */
+enum efivar_states
```

Comment:
I know it was I who suggested the enum, but now that I see how it is used (`eospayg_efi_secureboot_setup_active () == EFIVAR_FALSE` at the only callsite) I wonder if it should in fact return a single boolean in a similar "oddly inverted" way to the function below, `eospayg_efi_securebootoption_disabled`.

Reply:
Either way is fine – it's just an idea.

Reply:
hmm ...  For returning a single boolean, I think it will be unclear, if there is no SecureBootSetup EFI variable.

Reply:
I don't necessarily disagree, I'm just saying that there is another function in this same file that collapses the three states (true, false, not found) into a boolean to reflect how it is used.